rasterio/_shim1.pyx

```diff
@@ -8,6 +10,29 @@ from rasterio import dtypes
 from rasterio.enums import Resampling
 
 cimport numpy as np
+from rasterio._err cimport exc_wrap_pointer
+
+
+cdef GDALDatasetH open_dataset(
+        object filename, int flags, object allowed_drivers, object open_options,
```

Comment:
I _think_ its possible for open options to be passed to `rasterio.open()`and quietly ignored in this function.  Raising a `GDALBehaviorChangeException()` seems appropriate.

Reply:
I'd like to do that in a separate PR, since I'm not familiar with enforcing GDAL validation of options.

Reply:
I'm not suggesting that Rasterio validate options, just that with `GDAL<2.0` this _should_ raise an error inside `rasterio.open()`, but instead the `assert` will fail.

```
import affine
import rasterio as rio


with rio.open('tests/data/RGB.byte.tif', GEOREF_SOURCES='NONE') as src:
    # At this point the user does not know that the open option has been ignored
    assert src.transform == affine.Affine.identitity(), "Now this assert is mysteriously failing."
```

Reply:
Oh yeah, I see what you mean now 👍